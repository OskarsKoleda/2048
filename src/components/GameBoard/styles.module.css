.gameBoard {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    gap: .3rem;
    padding: 1rem;
    border-radius: 4px;
    background-color: rgba(127, 169, 34, 0.44);
}
