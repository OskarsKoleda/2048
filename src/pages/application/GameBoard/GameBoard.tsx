import React from 'react';
import Square from '../../../components/Square/Square';
import styles from './styles.module.css';

interface GameBoardProps {
  board: number[][];
}

const GameBoard = ({ board }: GameBoardProps) => {
  return (
    <div className={styles.gameBoard}>
      {board.flat().map((num, index) => (
        <Square key={`${num}-${index}`} squareValue={num} />
      ))}
    </div>
  );
};

export default React.memo(GameBoard);
