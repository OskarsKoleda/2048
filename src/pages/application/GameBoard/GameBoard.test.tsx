import { render } from '@testing-library/react';
import GameBoard from './GameBoard';

const props = {
  board: [
    [0, 2, 0, 2],
    [4, 0, 4, 0],
    [0, 8, 0, 8],
    [16, 0, 16, 0],
  ],
};

describe('GameBoard', () => {
  it('renders correctly', () => {
    const { container } = render(<GameBoard {...props} />);

    expect(container).toMatchSnapshot();
  });
});
