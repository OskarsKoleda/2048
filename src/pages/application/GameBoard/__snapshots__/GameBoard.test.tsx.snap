// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameBoard > renders correctly 1`] = `
<div>
  <div
    class="_gameBoard_6ae2ef"
  >
    <div
      class="_square_59a001 _value0_59a001"
    >
      0
    </div>
    <div
      class="_square_59a001 _value2_59a001"
    >
      2
    </div>
    <div
      class="_square_59a001 _value0_59a001"
    >
      0
    </div>
    <div
      class="_square_59a001 _value2_59a001"
    >
      2
    </div>
    <div
      class="_square_59a001 _value4_59a001"
    >
      4
    </div>
    <div
      class="_square_59a001 _value0_59a001"
    >
      0
    </div>
    <div
      class="_square_59a001 _value4_59a001"
    >
      4
    </div>
    <div
      class="_square_59a001 _value0_59a001"
    >
      0
    </div>
    <div
      class="_square_59a001 _value0_59a001"
    >
      0
    </div>
    <div
      class="_square_59a001 _value8_59a001"
    >
      8
    </div>
    <div
      class="_square_59a001 _value0_59a001"
    >
      0
    </div>
    <div
      class="_square_59a001 _value8_59a001"
    >
      8
    </div>
    <div
      class="_square_59a001 _value16_59a001"
    >
      16
    </div>
    <div
      class="_square_59a001 _value0_59a001"
    >
      0
    </div>
    <div
      class="_square_59a001 _value16_59a001"
    >
      16
    </div>
    <div
      class="_square_59a001 _value0_59a001"
    >
      0
    </div>
  </div>
</div>
`;
